marshal.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package httprequest
+
+import (
+	"bytes"
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+
+	"github.com/julienschmidt/httprouter"
+	"gopkg.in/errgo.v1"
+)
+
+// Marshal takes the input structure and creates an http request.
```

Comment:
// If url path contains a placeholder, but the input struct does not marshal
// to that placeholder, an error is raised.